virttest/qemu_vm.py

```diff
@@ -896,25 +896,35 @@ def add_memorys(devices, params):
 
             :param devices: VM devices container
             """
-            options = []
+            options, devs = [], []
+            params.setdefault("mem", "512")
             mem_params = params.object_params("mem")
-            if mem_params.get("maxmem"):
+            mem_params.setdefault("automem", "yes")
+            automem = mem_params["automem"] == "yes"
+            if automem:
```

Comment:
request memory size bigger than free memory size + 256M or request memory bigger than maxmen size, will reset it to avoid create VM failed.